File: go/leetcode/01266.go
package main

func intAbs(val int) int {
	if val < 0 {
		return val * -1
	}
	return val
}

func minTimeToVisitAllPoints(points [][]int) int {
	distance := 0

	for i := 1; i < len(points); i++ {
		x1, y1 := points[i-1][0], points[i-1][1]
		x2, y2 := points[i][0], points[i][1]

		distance += max(intAbs(x2-x1), intAbs(y2-y1))
	}
	return distance
}

File: python/LeetCode/01266.py
from typing import List

def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
    distance = 0
    n = len(points)

    for i in range(1, n):
        prev_x, prev_y = points[i - 1]
        new_x, new_y = points[i]

        dx, dy = abs(prev_x - new_x), abs(prev_y - new_y)

        diag = min(dx, dy)
        distance += diag + (dx - diag) + (dy - diag)

    return distance
